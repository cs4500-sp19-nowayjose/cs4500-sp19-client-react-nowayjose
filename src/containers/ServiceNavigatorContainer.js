import React from 'react'
import ServiceCategoryService from '../services/ServiceCategoryService'
import ServiceNavigator from '../components/ServiceNavigator/ServiceNavigator';

class ServiceNavigatorContainer extends React.Component {
    constructor(props) {
        super(props)
        this.serviceCategoryService = ServiceCategoryService.getInstance()
        this.state = {
            serviceCategories: [],
        }
    }
    componentDidMount() {
        this.serviceCategoryService
            .findAllServiceCategories()
            .then(serviceCategories => {
                var count = 0; 
                for(var i = 0; i<serviceCategories.length; i++) { 
                    var category = serviceCategories[i]
                    this.serviceCategoryService.findAllServicesForCategory(category.id)
                        .then(services => {
                            serviceCategories[count].services = services
                            count = count + 1
                            if (count == serviceCategories.length) {
                                this.setState({serviceCategories : serviceCategories})
                            }
                            
                    })
                }
            }
        )
    }

    render() {
        return (
            <ServiceNavigator
                serviceCategories={this.state.serviceCategories}
            />
        )
    }
}

export default ServiceNavigatorContainer