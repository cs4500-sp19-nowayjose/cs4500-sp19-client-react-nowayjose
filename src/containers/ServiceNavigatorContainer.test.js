import React from 'react'
import ServiceNavigatorContainer from './ServiceNavigatorContainer'
import TestRenderer from 'react-test-renderer';

test('[ServiceNavigatorContainer renders correctly]', () => {
    const testRenderer = TestRenderer.create(
        <ServiceNavigatorContainer/>)
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})

test('[ServiceNavigatorContainer renders correctly after selecting different category]', () => {
    const testRenderer = TestRenderer.create(
        <ServiceNavigatorContainer/>)
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance   = testRenderer.root
    let   selectCategory    = testInstance.findAllByProps({className: 'service-categories list-group-item no-border'})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})