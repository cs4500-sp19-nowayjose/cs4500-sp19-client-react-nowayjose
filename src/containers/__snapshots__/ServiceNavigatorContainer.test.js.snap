// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ServiceNavigatorContainer renders correctly after selecting different category] 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <ul
        className="list-group"
      />
    </div>
    <div
      className="col-9"
    >
      <ul
        className="list-group no-border"
      />
    </div>
  </div>
</div>
`;

exports[`[ServiceNavigatorContainer renders correctly after selecting different category] 2`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <ul
        className="list-group"
      />
    </div>
    <div
      className="col-9"
    >
      <ul
        className="list-group no-border"
      />
    </div>
  </div>
</div>
`;

exports[`[ServiceNavigatorContainer renders correctly] 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <ul
        className="list-group"
      />
    </div>
    <div
      className="col-9"
    >
      <ul
        className="list-group no-border"
      />
    </div>
  </div>
</div>
`;

exports[`[ServiceNavigatorContainer renders correctly] 2`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <ul
        className="list-group"
      />
    </div>
    <div
      className="col-9"
    >
      <ul
        className="list-group no-border"
      />
    </div>
  </div>
</div>
`;
